src/plugins/discover/public/utils/get_sharing_data.ts

```diff
@@ -56,10 +56,18 @@ export async function getSharingData(
 
   return {
     getSearchSource: (absoluteTime?: boolean): SerializedSearchSourceFields => {
+      const absoluteFilter = data.query.timefilter.timefilter.createFilter(index);
       const timeFilter = absoluteTime
-        ? data.query.timefilter.timefilter.createFilter(index)
+        ? absoluteFilter
         : data.query.timefilter.timefilter.createRelativeFilter(index);
 
+      // remove timeFilter from existing filter
+      if (Array.isArray(existingFilter)) {
+        existingFilter.filter((current) => isEqualFilters(current, absoluteFilter));
```

Comment:
The array `.filter` method does not mutate the original array, is this line intended to be like so:

```suggestion
        existingFilter = existingFilter.filter((current) => isEqualFilters(current, absoluteFilter));
```

Reply:
![image](https://user-images.githubusercontent.com/82822460/150109300-9e5d2084-0947-43e8-b2c4-a52593088f40.png)

Reply:
Array assignment has been mistakenly removed while refactoring, thank you for this catch.

Reply:
Note: since no test failed because of this change, we should have test cases here to catch it